File: Miscellaneous/Golang/125. Valid Palindrome.go
package main

import (
	"fmt"
	"regexp"
	"strings"
)

func main() {

	str := "A man, a plan, a canal: Panama"
	result := isPalindrome(str)
	fmt.Println(result)
}

func isPalindrome(s string) bool {
	reg, _ := regexp.Compile("[^a-zA-Z0-9]+")
	str := reg.ReplaceAllString(s, "")
	str = strings.ToLower(str)

	n := len(str)

	for i := 0; i < n; i++ {
		if str[i] != str[n-1-i] {
			return false
		}
	}

	return true
}

File: Miscellaneous/Python/125. Valid Palindrome.py
class Solution:
    def isPalindrome(self, s):
        string = ''.join(filter(str.isalnum, s))
        string = string.lower()
        n = len(string)

        for i in range(n):
            if string[i] != string[n-1-i]:
                return False

        return True

obj = Solution()
s = "A man, a plan, a canal: Panama"
ans = obj.isPalindrome(s)
print(ans)